### PR TITLE
Adding last modified to Dataset object #51

### PR DESCRIPTION
**Pull request by soumyasmruti**
Adding lastModified field to Dataset object along with tests. 
Closes: https://github.com/MarquezProject/marquez-java/issues/51

### CODE REVIEW

File: src/test/java/marquez/client/MarquezClientTest.java [hunk 1]
```diff
@@ -321,6 +340,36 @@ public void testGetDbTable() throws Exception {
...
+            .build();
+
+    final String modifiedMetaAsJson = JsonGenerator.newJsonFor(modifiedMeta);
+    final String modifiedDbTableAsJson = JsonGenerator.newJsonFor(DB_TABLE_MODIFIED);
+    when(http.put(url, modifiedMetaAsJson)).thenReturn(modifiedDbTableAsJson);
+
+    final Dataset modifiedDataset = client.createDataset(DB_TABLE_NAME, modifiedMeta);
+    assertThat(modifiedDataset).isInstanceOf(DbTable.class);
+    assertThat((DbTable) modifiedDataset).isEqualTo(DB_TABLE_MODIFIED);
+    assertThat(modifiedDataset.getLastModified()).isNotNull();
```
Comment (wslulciuc) [inline src/test/java/marquez/client/MarquezClientTest.java]: Given that you check for object equality:

```java
assertThat((DbTable) modifiedDataset).isEqualTo(DB_TABLE_MODIFIED);
```

I think we can remove the assertion:

```java
assertThat(modifiedDataset.getLastModified()).isNotNull();
```

or assert that `Dataset.lastModified` is after a timestamp `beforeModified`. For example:

```java
final Instant beforeModified = Instant.now();
// test logic
assertThat(modifiedDataset.getLastModified()).isAfter(beforeModified);

```

File: src/main/java/marquez/client/models/Dataset.java [hunk 2]
```diff
@@ -42,6 +42,7 @@
   @Getter @NonNull private final Instant createdAt;
   @Getter @NonNull private final Instant updatedAt;
   @Getter @NonNull private final String sourceName;
+  @Getter @Nullable private final Instant lastModified;
```
Comment (wslulciuc) [inline src/main/java/marquez/client/models/Dataset.java]: Given that `Dataset.lastModified` is nullable, we'll want to override the getter method and wrap the value in a `Optional`:

```java
  public Optional<Instant> getLastModified() {
    return Optional.ofNullable(lastModified);
  }
```

File: src/test/java/marquez/client/models/JsonGenerator.java [hunk 3]
```diff
@@ -121,6 +121,11 @@ private static String newJsonFor(final DbTable dbTable) {
         .put("createdAt", ISO_INSTANT.format(dbTable.getCreatedAt()))
         .put("updatedAt", ISO_INSTANT.format(dbTable.getUpdatedAt()))
         .put("sourceName", dbTable.getSourceName())
+        .put(
+            "lastModified",
+            dbTable.getLastModified() != null
```
Comment (wslulciuc) [inline src/test/java/marquez/client/models/JsonGenerator.java]: By wrapping `Dataset.lastModified` in an `Optional`, we can simplify the `null` check to:

```java
dbTable. getLastModified().map(ISO_INSTANT::format).orElse(null)
```

File: src/test/java/marquez/client/MarquezClientTest.java (hunk 1 above)
Comment (soumyasmruti) [reply src/test/java/marquez/client/MarquezClientTest.java]: Done

File: src/main/java/marquez/client/models/Dataset.java (hunk 2 above)
Comment (soumyasmruti) [reply src/main/java/marquez/client/models/Dataset.java]: Done

File: src/test/java/marquez/client/models/JsonGenerator.java (hunk 3 above)
Comment (soumyasmruti) [reply src/test/java/marquez/client/models/JsonGenerator.java]: Done

**Pull request merged**